debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnector.java

```diff
@@ -167,6 +171,7 @@ public void stop() {
         try {
             logger.info("Stopping MongoDB connector");
             this.config = null;
+            if (replicaSetMonitorExecutor != null) replicaSetMonitorExecutor.shutdown();
             try {
                 if ( this.monitorThread != null ) this.monitorThread.shutdown();
```

Comment:
Is this still needed with the executor?

Reply:
Yes, it is. The thread task does not return from `run` method until it receive an external signal for graceful termination.

Reply:
Understood, but the request above, `replicaSetMonitorExecutor.shutdown()` is this signal (as it's the executor running this thread). I think `ReplicaSetMonitorThread.shutdown()` should be removed and replaced by correct reaction to thread interruption in its `run()` method. I.e. if it detects an interruption, it should abort its run loop. Then we won't even have to keep a reference to `monitorThread` here.